### DISCUSSION TITLE
Add the option to sync zen mods with firefox sync

### DISCUSSION THREAD

**Discussion by AdilDevStuff** [zen-browser/desktop#1956] [Ideas]
Currently, reinstalling zen browser removes all zen mods and its required to install them again each time you reinstall zen. It would be really great to get some kind of sync option for zen mods as well so they get saved, preferably with firefox sync. As most people, I believe use firefox sync, it would be a great idea to implement sync of zen mods with firefox sync. Or any other sync option would work too (if you guys have already thought of any!)

Thanks!

**Comment by universish** [comment]
It is mozilla that will do this, not zen. 

If you want to back up zen mods, put the folders with the mods in your cloud account.
Otherwise to synchronize it zen browser would need to provide data server service. which is costly. This project is open source and is having trouble finding sponsors. So setting up this data server and setting up servers to create zen accounts is not possible for now.

**Reply by AdilDevStuff** [reply]
Yea, i totally get it!

**Reply by Eveeifyeve** [reply]
Maybe potentially a self hosted solution?

**Reply by Eragonfr** [reply]
That's technically false.

FF Sync is able to handle customs collections, which mean that you can store data in FF Sync even if that data wasn't specified by Firefox first. (source, myself and https://mozilla-services.readthedocs.io/en/latest/storage/apis-1.5.html#collections )

Zen could store the name/id and version and settings of each mod. And upon sync the browser download the mods from the mod store and restore their settings.

**Reply by mirsella** [reply]
if thats possible that would be ideal

**Comment by mirsella** [comment]
Hello !

Few db cloud provider offer a free plan that could maybe suffice for this use case ? Storing the ids of a few zen mods is very small.
for example https://turso.tech/pricing offer 9gb free, i think that's actually a lot of headroom for this use case :).
or other like neon or surrealdb (cloud beta) which offer 1gb, which should also suffice.
there's probably more free db services possible to use, i just listed those that i know of.

another solution which would require 0 ownership from zen, is using drive services as backup. i've seen a few extensions like stylus or tampermonkey, allow to sync their settings through for example google drive, with oauth.

i think this feature would be really useful. personally, i don't want to invest too much time looking at zen mods because i would need to find them all one by one again, test them, forget one, add a different one, inconsistent setup between devices, etc.
this feature would allow for a far better time using zen mods, and so using zen in general as zen mods can be really nice.

thanks

**Reply by universish** [reply]
Perhaps a torrent-like system with a blockchain structure can be implemented to use users' computers as servers, and this service can be provided. The server space can be divided between users and file sizes can be limited. We can ensure data security by mapping mode data configuration files to usernames and hosting this data encrypted.

**Comment by sandenbergmelo** [comment]
Any progress?

**Reply by Eveeifyeve** [reply]
I mean we are thinking ff sync, db provider or self hosted solution, but there is no offical decision yet.

**Comment by privatepenguinzero** [comment]
Any update in this matter?

**Comment by MrScottyTay** [comment]
FF Sync would be great. I just spent the better halves of two days configuring Zen exactly how I wanted, then installed it on a second device to find I have to do it all again, including all of the pins and essentials I had set up for each workspace and container... Massive bummer.
